connectivity/check/test.go

```diff
@@ -245,6 +248,8 @@ func (t *Test) Run(ctx context.Context) error {
 		s.Run(ctx, t)
 	}
 
+	t.completionTime = time.Now()
```

Comment:
The `completionTime` field won't be populated in case of an error in the loop.
How about moving this line into the `defer` block above?

Reply:
Good catch, we need to set it before returning an error, there is not in setting up the test as well